File: 2019W35/preppindataw36.R
library(dplyr)
library(tidyr)
library(readr)
library(stringr)

convert_ht <- function(df){
  df %>%
    mutate('ft'= as.integer(str_extract(`HT`,'^(\\d+)')),
           'inch' = as.integer(str_extract(`HT`,'\\s(\\d+)')),
           'Height (m)' = round((`ft`*12+`inch`)*2.54/100,2)) %>%
    select(-c('HT','ft','inch'))
}  

convert_wt <- function(df){
  df %>%
    mutate('lbs'= as.integer(str_extract(`WT`,'^(\\d+)')),
           'Weight (KGs)' = round(`lbs`*0.453592,2)) %>%
    select(-c('WT','lbs'))
}  

jersey_no <- function(df){
  df %>%
    mutate('NAME' = str_replace(`NAME`,'(\\d+)',',\\1')) %>%
    separate(`NAME`,c('NAME','Jersey Number'),sep=',',convert=TRUE)
}  

proc_all <- function(df){
  df %>% convert_ht(.) %>% convert_wt(.) %>% jersey_no(.)
}

spurs <- read_csv("E:/PD - Week 36 - Data.csv") %>% proc_all(.) 
nets <- read_csv("E:/PD - Week 36 - Brooklyn Data.csv") %>% proc_all(.) 

View(spurs)
View(nets)

File: 2019/2019W36/PreppindataW36.py
import pandas as pd
import re

def convert_ht(df):
    df['ft'] = df['HT'].apply(lambda x: int(re.search(r"^(\d+)'",x).group(1)))
    df['inch'] = df['HT'].apply(lambda x: int(re.search(r'\s(\d+)"',x).group(1)))
    df['Height (m)'] = df.apply(lambda x: round((x['ft']*12+x['inch'])*2.54/100,2), axis=1)
    df.drop(['HT','ft','inch'], axis=1, inplace=True)
    
def convert_wt(df):
    df['Weight (KGs)'] = df['WT'].apply(lambda x: round(int(re.search(r'^(\d+)',x).group(1))*0.453592,2))
    df.drop(['WT'], axis=1, inplace=True)    
    
def jersey_no(df):
    df['NAME'] = df['NAME'].apply(lambda x: re.sub('(\d+)',',\\1',x))     
    temp = df['NAME'].str.split(',', n = 1, expand = True) 
    df['NAME'] = temp[0]
    df['Jersey Number'] = temp[1].astype('int')
    
def proc_all(df):
    convert_ht(df)
    convert_wt(df)
    jersey_no(df)
    
spurs = pd.read_csv("E:\PD - Week 36 - Data.csv")
nets = pd.read_csv("E:\PD - Week 36 - Brooklyn Data.csv")

proc_all(spurs)
proc_all(nets)
